src/main/java/org/jenkinsci/plugins/mesos/JenkinsScheduler.java

```diff
@@ -518,6 +526,7 @@ private void detectAndAddExternalContainerInfo(Request request, CommandInfo.Buil
                 .newBuilder()
```

Comment:
This is not working with offers of different roles, e.g. `cpus(jenkins):5;mem(*):10240`.
I changed the method to reflect this kind of situations:

      private TaskInfo.Builder getTaskInfoBuilder(Offer offer, Request request, TaskID taskId, CommandInfo.Builder commandBuilder) {
          TaskInfo.Builder builder = TaskInfo.newBuilder()
            .setName("task " + taskId.getValue())
            .setTaskId(taskId)
            .setSlaveId(offer.getSlaveId())
                  .setCommand(commandBuilder.build());
    
          double cpusNeeded = request.request.cpus;
          double memNeeded = request.request.mem;
    
          for (Resource r : offer.getResourcesList()) {
              if (r.getName().equals("cpus") && cpusNeeded > 0) {
                  double cpus = Math.min(r.getScalar().getValue(), cpusNeeded);
                  builder.addResources(
                          Resource
                                  .newBuilder()
                                  .setName("cpus")
                                  .setType(Value.Type.SCALAR)
                                  .setRole(r.getRole())
                                  .setScalar(
                                          Value.Scalar.newBuilder()
                                                  .setValue(cpus).build()).build());
                  cpusNeeded -= cpus;
              } else if (r.getName().equals("mem") && memNeeded > 0) {
                  double mem = Math.min(r.getScalar().getValue(),memNeeded);
                  builder.addResources(
                          Resource
                                  .newBuilder()
                                  .setName("mem")
                                  .setType(Value.Type.SCALAR)
                                  .setRole(r.getRole())
                                  .setScalar(
                                          Value.Scalar.newBuilder()
                                                  .setValue(mem).build()).build());
                  memNeeded -= mem;
              } else if (cpusNeeded == 0 && memNeeded == 0) {
                  break;
              }
          }
          return builder;
      }